spark/src/main/java/org/apache/zeppelin/spark/SparkInterpreter.java

```diff
@@ -1278,6 +1281,7 @@ public void close() {
     if (numReferenceOfSparkContext.decrementAndGet() == 0) {
       sc.stop();
       sc = null;
+      sparkSession = null;
       if (classServer != null) {
```

Comment:
Good catch, when sparkSession is not null (spark 2.0), sparkSession.stop() should be called first.